romtools/trial_space_utils/svd_method_of_snapshots.py

```diff
@@ -47,12 +47,11 @@
 try:
     from mpi4py import MPI
     comm = MPI.COMM_WORLD
-    mpi_rank = comm.Get_rank()
-    num_processes = comm.Get_size()
 except ModuleNotFoundError:
     print("module 'mpi4py' is not installed")
-    mpi_rank = 0
-    num_processes = 1
+
+mpi_rank = 0
+num_processes = 1
```

Comment:
why are we doing this?

Reply:
It was throwing this error:
```
romtools/trial_space_utils/svd_method_of_snapshots.py:101:7: E0601: Using variable 'num_processes' before assignment (used-before-assignment)
romtools/trial_space_utils/svd_method_of_snapshots.py:106:7: E0601: Using variable 'mpi_rank' before assignment (used-before-assignment)
```

By bringing them into the global scope it resolved all errors/warnings. If I understand the code correctly, the functions will redefine these variables later if MPI is being used.

Reply:
i personally do not like global variables. can we change the code to avoid global?

Reply:
Yes let me try other solutions

Reply:
yes it is an impl function anyway so lets do that

Reply:
can you please do this in a spearate issue

Reply:
https://github.com/Pressio/rom-tools-and-workflows/pull/60